scripts/targets.cmake

```diff
@@ -0,0 +1,74 @@
+# ======================================================================
+#               CMake Targets Configuration
+# ======================================================================
+# This file defines all custom targets for the dmod-boot project
+# including install-firmware, connect, debug, monitor, and monitor-gdb
+
+# ======================================================================
+#               GDB Init Script Configuration
+# ======================================================================
+# Generate GDB init script from template with target-specific variables
+# Use QEMU-specific template in QEMU mode, otherwise use hardware template
+if(DMBOOT_QEMU)
+    set(GDB_INIT_TEMPLATE "${CMAKE_CURRENT_SOURCE_DIR}/configs/gdb/gdb_init_qemu.gdb.in")
+else()
+    set(GDB_INIT_TEMPLATE "${CMAKE_CURRENT_SOURCE_DIR}/configs/gdb/gdb_init.gdb.in")
+endif()
+
+configure_file(
+    ${GDB_INIT_TEMPLATE}
+    ${CMAKE_BINARY_DIR}/gdb_init.gdb
+    @ONLY
+)
+
+# ======================================================================
+#               Firmware Installation and Connection Targets
+# ======================================================================
+
+if(DMBOOT_QEMU)
+    message(STATUS "QEMU mode enabled - targets will use QEMU instead of OpenOCD")
+    
+    # In QEMU mode, install-firmware just copies the firmware to a known location
+    add_custom_target(install-firmware
```

Comment:
myślę, że to też spokojnie można sparametryzować - definiować target raz, a tylko na początku ustalić, że komenda instalacji firmware'u jest inna dla qemu